projects/kit/components/input-phone-international/input-phone-international.template.html

```diff
@@ -1,59 +1,67 @@
-            [focusable]="focusable"
-            [editable]="false"
-            [iconContent]="arrow"
-            [pseudoFocused]="open || null"
-            [readOnly]="readOnly"
-            [tuiHintContent]="null"
-        ></tui-primitive-textfield>
-        <tui-input-phone
-            class="input-phone tui-group__auto-width-item"
-            tuiTextfieldAutocomplete="off"
-            [tuiTextfieldLabelOutside]="true"
-            [formControl]="control"
-            [focusable]="focusable"
-            [countryCode]="inputPhoneCountryCode"
-            [phoneMaskAfterCountryCode]="phoneMaskAfterCountryCode"
-            [readOnly]="readOnly"
-            [pseudoInvalid]="pseudoInvalid"
-            [pseudoFocused]="pseudoFocused"
-            [pseudoHovered]="pseudoHovered"
-        >
-            {{inputPhoneCountryCode}}
-        </tui-input-phone>
-    </div>
-</tui-hosted-dropdown>
+<ng-container *tuiLet="countriesNames$ | async as countriesNames">
+    <tui-hosted-dropdown
+        class="hosted-dropdown"
+        [content]="dropdown"
+        [canOpen]="!readOnly"
+        [(open)]="open"
+    >
+        <div tuiGroup>
+            <tui-primitive-textfield
+                tuiHostedDropdownHost
+                class="country-select tui-group__auto-width-item"
+                [tuiTextfieldCustomContent]="countryValueContent"
+                [tuiTextfieldLabelOutside]="true"
+                [disabled]="disabled"
+                [focusable]="focusable"
+                [editable]="false"
+                [iconContent]="arrow"
+                [pseudoFocused]="open || null"
+                [readOnly]="readOnly"
+                [tuiHintContent]="null"
+            ></tui-primitive-textfield>
+            <tui-input-phone
+                class="input-phone tui-group__auto-width-item"
+                tuiTextfieldAutocomplete="off"
+                [tuiTextfieldLabelOutside]="true"
+                [formControl]="control"
+                [focusable]="focusable"
+                [countryCode]="inputPhoneCountryCode"
+                [phoneMaskAfterCountryCode]="phoneMaskAfterCountryCode"
+                [readOnly]="readOnly"
+                [pseudoInvalid]="pseudoInvalid"
+                [pseudoFocused]="pseudoFocused"
+                [pseudoHovered]="pseudoHovered"
+            >
+                {{inputPhoneCountryCode}}
+            </tui-input-phone>
+        </div>
+    </tui-hosted-dropdown>
 
-<ng-template #dropdown>
-    <tui-data-list>
-        <button
-            *ngFor="let item of countries"
-            tuiOption
-            (click)="onItemClick(item)"
-        >
-            <img class="country-item-flag" alt="" [src]="getFlagPath(item)" />
-            <span class="country-item-name">{{getCountryName(item)}}</span>
-            <span class="country-item-code">{{isoToCountryCode(item)}}</span>
-        </button>
-    </tui-data-list>
-</ng-template>
+    <ng-template #dropdown>
```

Comment:
Let's move templates under `tui-hosted-dropdown`. This way they will be under it in the DI tree (which technically is more correct, even though there are probably no benefits in this case) and we can the put `*tuiLet` on hosted dropdown and not introduce `ng-container` so there's less visual noise/nesting.

Reply:
done